File: 백준/code/15649.py
# ?? 15649

def sequence(k, used, tmp):
    if k == M:
        print(*tmp); return
    for i in range(N):
        if used & (1<<i): continue
        sequence(k + 1, used|(1<<i), tmp+str(arr[i]))

N, M = map(int, input().split())
arr = list(range(1, N + 1))
sequence(0, 0, '')

File: 백준/code/15649.js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split(' ').map(Number)

solution(...input)

function solution(N, M) {
  const numArr = new Array(N).fill(0).map((_,i) => i + 1)
  const permutationArr = permutation(numArr, M)
  for (const nums of permutationArr) {
    console.log(...nums)
  }
}

function permutation(arr, count) {
  let result = []
  let tmp = []

  permu(0)

  function permu(used) {

    if (tmp.length === count) {
      result.push([...tmp])
      return
    }

    for (let i = 0; i < arr.length; i++) {
      if (used & (1 << i)) continue
      tmp.push(arr[i])
      permu(used | (1 << i))
      tmp.pop()
    }
  }
  return result
}